### PR TITLE
[TS] LPS-87539

### PR DESCRIPTION
**Pull request by ChrisKian**
https://issues.liferay.com/browse/LPS-87539

CC @rgusztus

Tested and confirmed ExpandoColumn, ExpandoTable, and SystemEvent entries are cleared with virtual host.

### CODE REVIEW

**Comment by ChrisKian** [comment]
ci:test:sf

**Comment by ChrisKian** [comment]
ci:test:relevant

**Comment by drewbrokke** [comment]
Just started reviewing :)

File: portal-impl/src/com/liferay/portal/service/base/CompanyLocalServiceBaseImpl.java [hunk 1]
```diff
@@ -1272,6 +1284,12 @@ protected void runSQL(String sql) {
 	@BeanReference(type = RoleFinder.class)
 	protected RoleFinder roleFinder;
 
+	@BeanReference(
+		type = com.liferay.portal.kernel.service.SystemEventLocalService.class
+	)
+	protected com.liferay.portal.kernel.service.SystemEventLocalService
+		systemEventLocalService;
+
```
Comment (drewbrokke) [inline portal-impl/src/com/liferay/portal/service/base/CompanyLocalServiceBaseImpl.java]: @ChrisKian these are going to get blown out on the next `ant build-service-portal`.  This kind of reference needs to be added to the `service.xml` like so: https://github.com/liferay/liferay-portal/blob/48cd71b35a2d3b66e88f47685be7186cb7c52075/portal-impl/src/com/liferay/portal/service.xml#L189

File: portal-impl/src/com/liferay/portal/service/impl/CompanyLocalServiceImpl.java [hunk 2]
```diff
@@ -1451,6 +1454,33 @@ public Void call() throws Exception {
...
+
+		// ExpandoColumn
+
+		DeleteExpandoColumnActionableDynamicQuery
+			deleteExpandoColumnActionableDynamicQuery =
+				new DeleteExpandoColumnActionableDynamicQuery(
+					company.getCompanyId());
+
+		deleteExpandoColumnActionableDynamicQuery.performActions();
+
```
Comment (drewbrokke) [inline portal-impl/src/com/liferay/portal/service/impl/CompanyLocalServiceImpl.java]: I'm not sure these should go here.  Generally we want to delete higher-level objects earlier in the sequence (before the company itself is unregistered).

If it's indeed ok to delete these things after the company is unregistered, we could also explore using a ModelListener to remove these items.

**Comment by drewbrokke** [comment]
Hey @rgusztus @ChrisKian I wasn't able to test the change properly because the method threw an exception during group deletion.  In any case, there are a couple of things that need to change before we can send this on.  Please see my comments above and resend.  I'd say let's also try to include a regression test case if it's a reasonable effort.

**Pull request closed without merging**